src/00/03-dist-module-metadata/proposal.adoc

```diff
@@ -0,0 +1,368 @@
+ecosystem.
+
+To perform any further innovation in the format we must first standardize this
+existing practice.
+
+[#scope]
+== Scope
+
+This RFC does not seek to innovate upon the work done in P3286 and current
+implementations, only document and formalize the facts as they exist in build
+systems and stdlibs today. To that end it reproduces the format described in
+P3286 exactly.
+
+This RFC neither adds nor substracts anything to the format as originally
+proposed, and is strongly opposed to any modifications. Further innovations
+should be deferred to future RFCs that update the version and/or revision of
+P3286 files appropriately.
+
+Version 1/Revision 1 of this format has gone gold, this RFC is ex post facto
+bookkeeping. Wording changes may be appropriate so long as they make no
+structural or semantic changes to the format.
+
+[#design]
+== Design
+
+=== Requirements
+
+This requirements summary is reproduced verbatim from P3286:
+
+* A build system should have a way to identify which modules are provided by a
+pre-built library.
+
+* Locating the metadata file:
+
+  ** For the Standard Library:
+    *** The build system should be able to query the toolchain (either the
+        compiler or relevant packaging tools) for the location of that
+        metadata file.
+
+  ** Other Libraries:
+    *** In the absence of stronger package management, in environments where
+        that is viable, the build system may infer the location of the metadata
+        based on link-line fragments (P2701R0){fn-2}.
+    *** If package management is present, that information can be gathered in
+        implementation-defined ways.
+
+  ** The path to the metadata file should be related to the input files that
+     are given to the linker. The expectation is that different builds of the
+     library may have different metadata files.
+
+* The contents of the metadata *must* include:
+  ** The “logical name”{fn-3} of the importable unit being provided.
+  ** The path to the primary source file for the importable unit.
+  ** Any additional include paths required to translate that particular
+     importable unit.
+  ** Any compiler definitions required to translate that particular importable
+     unit.
+  ** Whether the module is a module provided by the standard library or not,
+     since those module names are reserved.
+
+* The contents of the metadata *may* include:
+  ** The “logical name” of importable units that are a dependency of that
+     translation unit.
+  ** Vendor-specific attributes.
+
+=== Format
+
+The file will be encoded in JSON{fn-4} and the data model is described in this
+paper as a JSON Schema{fn-5}. As it happens for P1689R5, the format will also
+require that file paths must be constrained to valid UTF-8 sequences.
+
+==== Schema
+
+For the information provided by the format, the following JSON Schema may be
+used.
+
+[source,javascript]
+----
+{
+  "$schema": "",
+  "$id": "http://example.com/root.json",
```

Comment:
```suggestion
  "$id": "http://example.com/root.json",
```

Will this get replaced with a real URI, perhaps published by this repo?

FYI @grafikrobot since you might have thoughts on how schemas are published when required.